probe-rs/src/architecture/arm/memory/mod.rs

```diff
@@ -1,21 +1,52 @@
 //! Types and functions for interacting with target memory.
 
-pub(crate) mod adi_v5_memory_interface;
+mod adi_v5_memory_interface;
 pub mod romtable;
 
+pub(crate) use adi_v5_memory_interface::ADIMemoryInterface;
+
 use crate::{memory::MemoryInterface, probe::DebugProbeError, CoreStatus};
 
 use super::{
-    ap::memory_ap::{registers, MemoryAp},
-    communication_interface::SwdSequence,
-    ArmError, ArmProbeInterface, DapAccess,
+    ap_v1, ap_v2, communication_interface::SwdSequence, ArmError, ArmProbeInterface, DapAccess,
+    FullyQualifiedApAddress,
 };
-pub use romtable::{Component, ComponentId, CoresightComponent, PeripheralType};
+pub use romtable::{Component, ComponentId, CoresightComponent, PeripheralType, RomTable};
+
+/// A generic status indication for an AP.
+pub enum Status {
+    /// A CSW associated with an APv1 (ADIv5) access port.
+    V1(ap_v1::memory_ap::registers::CSW),
+    /// A CSW associated with an APv2 (ADIv6) access port.
+    V2(ap_v2::registers::CSW),
```

Comment:
Is there a benefit to knowing if a CSW register is from an APv1 or an APv2? It seems that they are the same, only the name of `CSW.SPIDEN` is changed to `CSW.SDeviceEn`.

Reply:
Spawned https://github.com/probe-rs/probe-rs/issues/3061 to track this one

Reply:
https://github.com/probe-rs/probe-rs/issues/3062